packages/client/src/runtime/core/engines/binary/BinaryEngine.ts

```diff
@@ -812,9 +812,9 @@ You very likely have the wrong "binaryTarget" defined in the schema.prisma file.
 
     if (action === 'start') {
       const jsonOptions = JSON.stringify({
-        max_wait: arg?.maxWait ?? 2000, // default
-        timeout: arg?.timeout ?? 5000, // default
-        isolation_level: arg?.isolationLevel,
+        max_wait: arg?.maxWait ?? this.config.transactionOptions.maxWait,
```

Comment:
Rather then copying this code into 3 engine classes, I'd suggest to do it in a single place in [`getPrismaClient`](https://github.com/prisma/prisma/blob/d212cd87d0e3586fde996f2c479226a3d53b22e1/packages/client/src/runtime/getPrismaClient.ts#L710).

Additionally, default isolation level should probably also apply to batch transactions (the ones that take in array), where this path handles only iTx (the ones that take callback).

The place to apply default to the batch transactions is [a couple of lines above](https://github.com/prisma/prisma/blob/d212cd87d0e3586fde996f2c479226a3d53b22e1/packages/client/src/runtime/getPrismaClient.ts#L689)

Reply:
maybe fixed by those commits.
https://github.com/prisma/prisma/pull/19592/commits/be405daa9315b71a51c55f987e85899176eef0fe
https://github.com/prisma/prisma/pull/19592/commits/e855ca6c05581e4e2069aba94fc9820105ce5a8d